### PR TITLE
Add `remove-checks-tab` feature

### LINKED ISSUE #1304: Hide "Checks" tab on Pull request page
`Checks` tab is not related directly to the pull request workflow, would be helpful to have a feature that will disable this tab.
image

### PR DESCRIPTION
**Pull request by janat08**
checks not gone- non zero: https://github.com/probot/semantic-pull-requests/pull/40
 not gone- selected: https://github.com/sindresorhus/refined-github/pull/1928/checks
 this page demonstrates it being gone unless youre owner

closes #1304

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you fix the merge conflict and rebase from master (we enabled [TS strict mode](https://github.com/sindresorhus/refined-github/commit/7b3b275ad9a1b1409c56310c0b8ab286bef1f75f)) and fix any issues?

File: source/features/remove-checks-tab.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	await safeElementReady(`
+		[data-hotkey="g 3"]
+	`); // Wait for the tab to be loaded
+
+	const checksTab = select([
+		'[data-hotkey="g 3"]'
+	].join());
+
+	// Only remove the tab if it's not the current page and if it has 0 projects
```
Comment (bfred-it) [inline source/features/remove-checks-tab.tsx]: Here it still mentions Projects

File: source/features/remove-checks-tab.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		return;
+	}
+
+	await safeElementReady(`
+		[data-hotkey="g 3"]
+	`); // Wait for the tab to be loaded
+
+	const checksTab = select([
+		'[data-hotkey="g 3"]'
+	].join());
```
Comment (bfred-it) [inline source/features/remove-checks-tab.tsx]: `join` don’t make sense here, don’t copy-paste code without reading it 😅 

You can add `const checkTabs =` before `await` and skip the `select` altogether

File: source/libs/page-detect.ts [hunk 3]
```diff
@@ -77,6 +77,8 @@ export const isRepo = (): boolean => /^[^/]+\/[^/]+/.test(getCleanPathname()) &&
 	!isGist() &&
 	!isRepoSearch();
 
+export const isOwnRepo = (): boolean => isRepo() && getUsername() === getOwnerAndRepo().ownerName;
```
Comment (bfred-it) [inline source/libs/page-detect.ts]: I think this can be tested in JSdom

File: source/features/remove-checks-tab.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	if (!checksTab.matches('.selected') && select('.Counter', checksTab).textContent.trim() === '0') {
+		checksTab.remove();
+	}
+}
+
+features.add({
+	id: 'remove-checks-tab',
+	include: [
+		features.isPR
+	],
```
Comment (bfred-it) [inline source/features/remove-checks-tab.tsx]: `isOwnRepo` should be listed here inside `exclude`

Comment (janat08) [reply source/features/remove-checks-tab.tsx]: ty

File: source/libs/page-detect.ts (hunk 3 above)
Comment (janat08) [reply source/libs/page-detect.ts]: getUsername depends on DOM

Comment (janat08) [reply source/libs/page-detect.ts]: It's basically a combination of two base functions. I don't see any jsdom tests to learn from.

Comment (bfred-it) [reply source/libs/page-detect.ts]: Indeed. The `is404` test in `tests/page-detect` technically also depends on the `<title>` tag, but it's more easily-accessible via `document.title`. Perhaps something like:

```js
test('isOwnRepo', t => {
	document.head.insertAdjacentHTML('<meta name="user-login" content="sindresorhus">');
	location.href = 'https://github.com/sindresorhus/refined-github/pull/1929'
	t.true(pageDetect.isOwnRepo());
});
```

The problem is that `isOwnRepo` is a one-time function, so unless @sindresorhus suggests a way to test it further, we can only do this once

Comment (bfred-it) [reply source/libs/page-detect.ts]: Actually this function is incorrect and should be dropped. This is what was in `remove-projects-tab`

```js
	// If there's a settings tab, the current user can disable the projects,
	// so the tab should not be hidden
	if (select.exists([
		'.js-repo-nav [data-selected-links^="repo_settings"]', // In repos
		'.pagehead-tabs-item[href$="/settings/profile"]' // In organizations
	].join())) {
		return;
	}
```

Which is different from `currentUser === repoOwner`. This code should also work in organizations where the owner doesn't match the `repoOwner` part of the URL, like https://github.com/xojs/xo

File: source/libs/page-detect.ts [hunk 5]
```diff
@@ -77,7 +77,9 @@ export const isRepo = (): boolean => /^[^/]+\/[^/]+/.test(getCleanPathname()) &&
 	!isGist() &&
 	!isRepoSearch();
 
-export const isRepoDiscussionList = (): boolean => /^(issues$|pulls$|labels\/)/.test(getRepoPath()!);
+export const isOwnRepo = (): boolean => isRepo() && getUsername() === getOwnerAndRepo().ownerName;
+
+export const isRepoDiscussionList = (): boolean => /^(issues$|pulls$|labels\/)/.test(getRepoPath());
```
Comment (bfred-it) [inline source/libs/page-detect.ts]: Unrelated change?

**Comment by bfred-it** [comment]
Please run this to delete the tags from your fork and locally, we keep pulling them from your fork and pushing them to this repo (we no longer use tags)

```sh
git tag -d $(git tag | grep -E '.')
git ls-remote --tags --refs origin | cut -f2 | xargs git push origin --delete
```

File: source/libs/page-detect.ts (hunk 3 above)
Comment (janat08) [reply source/libs/page-detect.ts]: basically, copy and pasted the selectors. I assume it applies exactly in the same way.

File: source/features/remove-checks-tab.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+	// so the tab should not be hidden
+	if (select.exists([
+		'.js-repo-nav [data-selected-links^="repo_settings"]', // In repos
+		'.pagehead-tabs-item[href$="/settings/profile"]' // In organizations
+	].join())) {
+		return false;
+	}
+
+	// Only remove the tab if it's not the current page and if it has 0 checks
+	const checksCounter = select('.tabnav-tab[href$="/checks"]:not(.selected) .Counter');
```
Comment (bfred-it) [inline source/features/remove-checks-tab.tsx]: I dropped `safeElementReady` here because `onAjaxedPages` always waits for `domReady` first.

We currently don't have a callback that is called before domready _and_ supports ajaxed pages. It'd be great to have that

**Comment by bfred-it** [comment]
Thanks! ✅
